install.sh

```diff
@@ -90,73 +87,63 @@ function curl () {
   # the 'curl: (7) Couldn't connect to server' error, a for loop is used
   # here.
   for (( i = 0; i < 4; i++ )); do
-    # force TLS as we know GitLab supports it
-    env -u LD_LIBRARY_PATH ${CURL} --ssl-reqd --tlsv1.2 -C - "${@}" && return 0
-    echo_info "Retrying, $((3-i)) retries left."
+    # Force TLS as we know GitLab supports it
+    if [ "$(env -u LD_LIBRARY_PATH "${CURL}" --ssl-reqd --tlsv1.2 -C - "${@}")" ]; then
+      echo_info "Retrying, $((3-i)) retries left."
+    else
+      return 0
+    fi
   done
   # The download failed if we're still here.
   echo_error "Download failed :/ Please check your network settings."
   return 1
 }
 
-case "${ARCH}" in
-"i686"|"x86_64"|"armv7l"|"aarch64")
-  LIB_SUFFIX=
-  # See https://superuser.com/a/1369875
-  # If /bin/bash is 64-bit, then set LIB_SUFFIX, as this is true on both
-  # x86_64 and aarch64 userspace
-  # shellcheck disable=SC2046
-  [ $(od -An -t x1 -j 4 -N 1  /bin/bash) == 02 ] && LIB_SUFFIX='64'
-  if [[ $LIB_SUFFIX == '64' ]] && [[ $ARCH == 'aarch64' ]]; then
-    echo_error "Your device is not supported by Chromebrew yet :/"
-    exit 1
-  fi
-  ;;
-*)
+# See https://superuser.com/a/1369875
+# If /bin/bash is 64-bit, then set LIB_SUFFIX, as this is true on both
+# x86_64 and aarch64 userspace
+# shellcheck disable=SC2046
+[ $(od -An -t x1 -j 4 -N 1  /bin/bash) == 02 ] && LIB_SUFFIX='64'
+if [[ $LIB_SUFFIX == '64' ]] && [[ $ARCH == 'aarch64' ]]; then
   echo_error "Your device is not supported by Chromebrew yet :/"
   exit 1
-  ;;
-esac
-
-echo_info "\n\nDoing initial setup for install in ${CREW_PREFIX}."
-echo_info "This may take a while if there are preexisting files in ${CREW_PREFIX}...\n"
-
-# This will allow things to work without sudo.
-crew_folders="bin cache doc docbook etc include lib lib$LIB_SUFFIX libexec man sbin share tmp var"
-for folder in $crew_folders
-do
-  if [ -d "${CREW_PREFIX}"/"${folder}" ]; then
-    echo_intra "Resetting ownership of ${CREW_PREFIX}/${folder}"
-    sudo chown -R "$(id -u)":"$(id -g)" "${CREW_PREFIX}"/"${folder}"
-  fi
-done
-sudo chown "$(id -u)":"$(id -g)" "${CREW_PREFIX}"
+fi
 
-# Delete ${CREW_PREFIX}/{var,local} symlinks on some Chromium OS distro
-# if they exist.
+# Delete ${CREW_PREFIX}/{var,local} symlinks which exist on some Chromium OS distros.
 [ -L ${CREW_PREFIX}/var ] && sudo rm -f "${CREW_PREFIX}/var"
 [ -L ${CREW_PREFIX}/local ] && sudo rm -f "${CREW_PREFIX}/local"
 
-# Prepare directories.
-for dir in "${CREW_CONFIG_PATH}/meta" "${CREW_DEST_DIR}" "${CREW_PACKAGES_PATH}" "${CREW_CACHE_DIR}" ; do
-  if [ ! -d "${dir}" ]; then
-    mkdir -p "${dir}"
+if [ -d ${CREW_PREFIX} ]; then
+  if [ "$(ls -A ${CREW_PREFIX})" ]; then
+    echo_error "${CREW_PREFIX} is not empty, please move all files you wish to keep out of ${CREW_PREFIX} then clear it."
+    exit 1
   fi
-done
+else
+  sudo mkdir ${CREW_PREFIX}
```

Comment:
This will generate an error since `/usr/local` is read-only.  It won't be needed anyway since you are prompting the user to clear out their `${CREW_PREFIX}` folder manually (not ideal).

Reply:
It won't generate an error if the user's `CREW_PREFIX` isin't `/usr/local`.

When this line comes into play is if a user selects a folder which does not exist yet, and in this case it will be created and the install will continue normally.

I thought it made more sense to tell the user to clear out their `CREW_PREFIX` folder manually rather than doing it silently and possibly deleting important files.

Reply:
But it will generate an error for the other 99.9% of users that don't relocate `CREW_PREFIX`.  Just sayin'.  It might make more sense to prompt the user with a warning and clear it out for them rather than have them exit and start the install again.  What I'm trying to say is it would be a deterrent and may antagonize new users to force them to do it manually.

Reply:
I do see what you're saying about clearing out `CREW_PREFIX`, would it make sense to give them an option to have the installer clear it out for them?

Reply:
Yes, that would be the most friendly thing to do.  With a big warning message prior to doing it, of course.